src/cpp/shared/game/game.h

```diff
@@ -221,6 +240,30 @@ extern int gbInCampaign;
 
 extern int gbGameOver;
 
+extern __int16 giUltimateArtifactBaseX;
+extern __int16 giUltimateArtifactBaseY;
+extern __int16 giUltimateArtifactRadius;
+
+extern char gpcBeforeGSpellLimits[];
+extern int giPlayerInitialResourcesHuman[NUM_DIFFICULTIES][NUM_RESOURCES];
+extern int giPlayerInitialResourcesAI[NUM_DIFFICULTIES][NUM_RESOURCES];
+
+extern int __fastcall FindLastToken(const char *a1, char a2);
```

Comment:
Ditto for this.

Reply:
Okay....if you're declaring them extern, I guess it's okay.

Reply:
What?

Reply:
I mean, if these were already in the original code, and you're leaving them in and not changing anything, then there's not much to be gained by replacing them with a standard library version.

Reply:
After further reading this comment, I see now that it was a development on the previous one, and that my last post on the previous comment is now obsolete.